File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/ScatterChart.swift
//
//  ScatterChart.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class ScatterChart: TKExamplesExampleViewController {
    
    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        for i in 0..<2 {
            // >> chart-scatter-swift
            var points  = [TKChartDataPoint]()
            for _ in 0..<20 {
                let randomX = Int(arc4random()%1450)
                let randomY = Int(arc4random()%150)
                points.append(TKChartDataPoint(x:randomX, y:randomY))
            }
            let series = TKChartScatterSeries(items:points)
            // << chart-scatter-swift
            series.title = String(format:"Series % d", (i+1))
            if(2==i){
                series.selection = TKChartSeriesSelection.DataPoint
            }
            else{
                series.selection = TKChartSeriesSelection.Series
            }
            series.style.paletteMode = TKChartSeriesStylePaletteMode.UseItemIndex
            series.marginForHitDetection = 300
            chart.addSeries(series)
        }
    
        chart.xAxis!.allowPan = true
        chart.xAxis!.allowZoom = true
        chart.yAxis!.allowPan = true
        chart.yAxis!.allowZoom = true
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

File: Examples/Examples/Docs/ChartTypes/ScatterChart.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("ScatterChart")]
	public class ScatterChart: XamarinExampleViewController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			TKChart chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			Random r = new Random ();
			for (int i = 0; i < 2; i++) {
				// >> chart-scatter-cs
				List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
				for (int j = 0; j < 20; j++) {

					list.Add(new TKChartDataPoint (new NSNumber (r.Next() % 1450), new NSNumber (r.Next () % 150)));
				}

				TKChartScatterSeries series = new TKChartScatterSeries (list.ToArray());
				// << chart-scatter-cs
				series.Title = string.Format ("Series: {0}", i + 1);
				if (2 == i) {
					series.SelectionMode = TKChartSeriesSelectionMode.DataPoint;
				} else {
					series.SelectionMode = TKChartSeriesSelectionMode.Series;
				}
				series.Style.PaletteMode = TKChartSeriesStylePaletteMode.UseItemIndex;
				series.MarginForHitDetection = 300;
				chart.AddSeries (series);
			}

			chart.XAxis.AllowZoom = true;
			chart.YAxis.AllowZoom = true;
		}
	}
}

